### PR TITLE
Don't show news posts from the future

### PR DESCRIPTION
**Pull request by cl8n**
osu! wiki wants to be able to merge news posts to the repo before they go live. is this all that is needed?

### CODE REVIEW

File: app/Models/NewsPost.php [hunk 1]
```diff
@@ -65,7 +65,7 @@ public static function lookupAndSync($slug)
 
         $post->sync();
 
-        if ($post->page !== null) {
+        if ($post->page !== null && $post->published_at !== null && $post->published_at <= Carbon::now()) {
```
Comment (nanaya) [inline app/Models/NewsPost.php]: use `->isPast()`

File: app/Models/NewsPost.php [hunk 2]
```diff
@@ -77,7 +77,7 @@ public static function pageVersion()
 
     public static function search($params)
     {
-        $query = static::whereNotNull('published_at');
+        $query = static::whereNotNull('published_at')->where('published_at', '<=', Carbon::now());
```
Comment (nanaya) [inline app/Models/NewsPost.php]: Can use a scope for this and `scopeDefault`.

Comment (cl8n) [reply app/Models/NewsPost.php]: something like "scopePublished" and then scopedefault is that + orderby?

Comment (nanaya) [reply app/Models/NewsPost.php]: yeah

**Pull request merged**